### PR TITLE
Fix bugs exposed by posix port

### PR DESCRIPTION
**Pull request by jepler**
While working on some experimental code ("posix port") I found some bugs & build problems.

The added includes are to get structure definitions and/or function prototypes for things that are included indirectly *only when certain other features are enabled*, which is why the problems weren't seen before.

The removed includes are for stuff not yet implemented on the POSIX port, where the include was apparently copied as boilerplate and is not actually needed.

### CODE REVIEW

File: shared-module/epaperdisplay/EPaperDisplay.h [hunk 1]
```diff
@@ -26,7 +26,6 @@
 
 #pragma once
 
-#include "shared-bindings/digitalio/DigitalInOut.h"
```
Comment (tannewt) [inline shared-module/epaperdisplay/EPaperDisplay.h:29]: I think this is causing build errors.

**Review by tannewt — CHANGES_REQUESTED** [review]
Is the random bits 0 behavior change to match CPython?

File: shared-module/epaperdisplay/EPaperDisplay.h (hunk 1 above)
Comment (jepler) [reply shared-module/epaperdisplay/EPaperDisplay.h:29]: yeah this part of the change isn't right. Instead I need to figure out why EPaperDisplay was being included where it wasn't enabled.

**Comment by tannewt** [comment]
Why don't we update black in #8906 and then merge that into main?

**Comment by jepler** [comment]
I'm fine with rebasing this once #8906 is merged, if that makes sense to you.

**Comment by tannewt** [comment]
Ok, merged in #8912

**Comment by jepler** [comment]
rebased.

**Review by tannewt — APPROVED** [review]
Thanks for the update!

**Pull request merged**